src/Core/Components/Switch/FluentSwitch.razor

```diff
@@ -0,0 +1,18 @@
+﻿@namespace Microsoft.FluentUI.AspNetCore.Components
+@using Microsoft.FluentUI.AspNetCore.Components.Extensions
+@inherits FluentInputBase<bool>
+
+<FluentField InputComponent="@this" ForId="@Id" Class="@ClassValue" Style="@((ReadOnly ? "pointer-events: none;" : "") + StyleValue)">
+    <fluent-switch aria-label="@AriaLabel"
+                   autofocus="@Autofocus"
+                   checked="@CurrentValue"
+                   disabled="@Disabled"
+                   id="@Id"
+                   name="@Name"
```

Comment:
You need to add a `tabindex="-1"` if `ReadOnly=true` and `!AdditionalAttributes?.ContainsKey("tabindex")`.
If this is not the case, the user can navigate to the component using the keyboard.